### ISSUE TITLE
The Altitude Anomaly (Tunnels / Bridges)

### ISSUE THREAD

**Issue by Hematot** [Project-OSRM/osrm-backend#1263]
When routing to/from a valid street address (street name and number) that happens to be on top of a tunnel or below a bridge, the address is ignored and the routing is made to the nearest point in/on the tunnel/bridge instead.

Here’s a route that shows an example with both a tunnel and a bridge snap:
http://osrm.at/a1E
Addresses entered as such:
Start: Madison Street 137, Manhattan
End: 1st Avenue 660, Manhattan

A couple of routes to show that this is not a local occurrence:
http://osrm.at/a1i
Start: Enrisvägen 2, Gothenburg
End: Alevägen 12, Gothenburg

http://osrm.at/a1j
Start: Strehlowweg 42, Hamburg
End: Strehlowweg 60, Hamburg

Funnily enough, when moving the marker graphically over tunnels/bridges:
1, the routing (Graphically as well as Route Description) snaps to the bridge/tunnel.
2, the Start and End fields prefferrably “snaps” to a street, more so if there are street numbers close.
In other words, there is a discrepancy between the Routing and From/To fields.

(P.S. I hope I'm reporting this bug in a correct way. I'm totally new to GitHub and really don't know how to do this, so please excuse me if this is the wrong way to do it...)
